automation/ci/scripts/manage_apps.sh

```diff
@@ -0,0 +1,228 @@
+#!/bin/bash
+
+set -euo pipefail
+set -x
+
+# Usage: manage_apps.sh --env_list <env_list> --env_state <env_state> --check_activity <check_activity> --cg_api <cg_api> --cg_org <cg_org>
+
+# Parse arguments
+while [[ "$#" -gt 0 ]]; do
+  case $1 in
+    --env_list) env_list="$2"; shift ;;
+    --env_state) env_state="$2"; shift ;;
+    --check_activity) check_activity="$2"; shift ;;
+    --activity_timeout) activity_timeout=$2; shift ;;
+    --cg_api) cg_api="$2"; shift ;;
+    --cg_org) cg_org="$2"; shift ;;
+    --branch) branch="$2"; shift ;;
+    --build) build="$2"; shift ;;
+    *) echo "Unknown parameter: $1"; exit 1 ;;
+  esac
+  shift
+done
+
+if [[ -z "${env_list:-}" || -z "${env_state:-}" || -z "${check_activity:-}" || -z "${cg_api:-}" || -z "${cg_org:-}" || -z "${branch:-}" || -z "${build:-}" ]]; then
+  echo "Error: Missing required arguments."
+  exit 1
+fi
+
+# Define prefixes for environments
+primary_prefix="tta-smarthub"
+secondary_prefixes=("tta-similarity-api")
+
+# Convert the comma-separated list into an array using substitution
+apps=(${env_list//,/ })
+
+for env in "${apps[@]}"; do
+  echo "Processing environment group: $env"
+
+  # Normalize environment name to lowercase for suffix
+  env_suffix=$(echo "${env}" | tr '[:upper:]' '[:lower:]')
+
+  # Dynamically derive variable names for environment
+  space_var="CLOUDGOV_${env^^}_SPACE"
+  username_var="CLOUDGOV_${env^^}_USERNAME"
+  password_var="CLOUDGOV_${env^^}_PASSWORD"
+
+  # Resolve the actual values of the variables
+  space="${!space_var:-}"
+  username="${!username_var:-}"
+  password="${!password_var:-}"
+
+  if [[ -z "$space" || -z "$username" || -z "$password" ]]; then
+    echo "Error: Missing required environment variable(s) for $env"
+    exit 1
+  fi
+
+  # Log in to Cloud Foundry
+  cf login \
+    -a "$cg_api" \
+    -u "$username" \
+    -p "$password" \
+    -o "$cg_org" \
+    -s "$space"
+
+  ./automation/ci/scripts/acquire-lock.sh \
+    "$env" \
+    "$branch" \
+    "$build"
+
+  # Perform activity check only for the primary prefix (tta-smarthub)
+  if [[ "$check_activity" == "true" && "$env_state" == "stop" ]]; then
+    app_name="${primary_prefix}-${env_suffix}"
+    
+    current_state=$(cf apps | grep "${app_name}" | awk '{print $2}' || echo "unknown")
+    if [ "$current_state" == "stopped" ]; then
+          echo "$app_name is already stopped."
+
+          ./automation/ci/scripts/release-lock.sh \
+            "$env" \
+            "$branch" \
+            "$build"
+      continue
+    fi
+
+    echo "Fetching recent logs for $app_name..."
+
+    # Safely fetch recent logs
+    recent_logs=$(cf logs --recent "$app_name" 2>/dev/null || echo "")
```

Comment:
I guess we'll have to be careful to only target env's with PII stripped data, as the SQL will be publicly accessible in CircleCi

Reply:
should a guard be put in to force this not to run on prod

Reply:
I don't want to hold this up, unless it is simple. If not maybe log a ticket to document the risk and implement in the future?

Reply:
will exit if ever tried to run on prod